radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/KeySignHandler.java

```diff
@@ -61,81 +61,76 @@
  * permissions under this License.
  */
 
-package com.radixdlt.api.node.account;
-
-import org.json.JSONArray;
-import org.json.JSONObject;
+package com.radixdlt.api.core.handlers;
 
 import com.google.inject.Inject;
-import com.radixdlt.api.service.SubmissionService;
-import com.radixdlt.api.util.ActionParser;
+import com.google.inject.Provider;
+import com.radixdlt.api.core.CoreJsonRpcHandler;
+import com.radixdlt.api.core.model.CoreApiException;
+import com.radixdlt.api.core.model.CoreModelMapper;
+import com.radixdlt.api.core.openapitools.model.PublicKeyNotSupportedError;
+import com.radixdlt.api.core.openapitools.model.KeySignRequest;
+import com.radixdlt.api.core.openapitools.model.KeySignResponse;
+import com.radixdlt.atom.TxLowLevelBuilder;
+import com.radixdlt.atom.Txn;
 import com.radixdlt.consensus.HashSigner;
 import com.radixdlt.consensus.bft.Self;
-import com.radixdlt.identifiers.AID;
-import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.engine.RadixEngine;
+import com.radixdlt.engine.parser.exceptions.TxnParseException;
 import com.radixdlt.qualifier.LocalSigner;
-import com.radixdlt.utils.functional.Result;
-
-import java.util.List;
-import java.util.Optional;
-
-import static com.radixdlt.api.util.JsonRpcUtil.jsonObject;
-import static com.radixdlt.api.util.JsonRpcUtil.optString;
-import static com.radixdlt.api.util.JsonRpcUtil.safeArray;
-import static com.radixdlt.api.util.JsonRpcUtil.toResponse;
-import static com.radixdlt.api.util.JsonRpcUtil.withRequiredParameters;
-import static com.radixdlt.utils.functional.Result.allOf;
+import com.radixdlt.statecomputer.LedgerAndBFTProof;
+import com.radixdlt.utils.Bytes;
 
-public class LocalAccountHandler {
-	private final AccountInfoService accountService;
-	private final SubmissionService submissionService;
-	private final ActionParser actionParserService;
+public final class KeySignHandler extends CoreJsonRpcHandler<KeySignRequest, KeySignResponse> {
```

Comment:
what's this api used for? seems quite dangerous.
I'd really like to have at least a separate flag that explicitly enables it (disabled by default).

Reply:
Hrm, good point. Api is used mainly for validator runners. But I think you're right, this should probably explicitly enabled.